remoteappmanager/static/js/admin/vue-components/UsersView.js

```diff
@@ -0,0 +1,221 @@
+define([
+  "jquery",
+  "components/vue/dist/vue",
+  "jsapi/v1/resources"
+], function($, Vue, resources) {
+  "use strict";
+
+  var NewUserDialog = {
+    template: `
+      <modal>
+        <div class="modal-header"><h4>Create New User</h4></div>
+        <div class="modal-body">
+          <vue-form :state="formstate" v-model="formstate" @submit.prevent="createNewUser">
+            <validate auto-label class="form-group required-field" :class="fieldClassName(formstate.name)">
+              <label class="control-label">User Name</label>
+              <input type="text" name="name" class="form-control" required v-model="model.name">
+
+              <field-messages name="name" show="$touched || $submitted">
+                <span class="help-block" slot="required">User Name cannot be empty</span>
+              </field-messages>
+            </validate>
+
+            <div class="modal-footer">
+              <button type="button" class="btn btn-default" @click="close()">Cancel</button>
+              <button class="btn btn-primary" type="submit" :disabled="formstate.$invalid">Submit</button>
+            </div>
+          </vue-form> 
+        </div>
+    </modal>`,
+    props: ['show'],
+    
+    data: function () {
+      return {
+        formstate: {},
+        model: {
+          name: ''
+        }
+      };
+    },
+    methods: {
+      close: function () {
+        this.$emit('closed');
+      },
+      fieldClassName: function (field) {
```

Comment:
should be in the template directly, it will reduce your code:
```html
:class="{ 'has-error': formstate.name && (formstate.name.$touched || formstate.name.$submitted) && formstate.name.$invalid }"
```
It's one long line vs 11 lines